transactor/src/main/scala/io/mediachain/transactor/Types.scala

```diff
@@ -213,6 +215,15 @@ object Types {
   
   // Datastore interface
   trait Datastore {
-    def put(obj: DataObject): Reference
+    def put(obj: Array[Byte], hash: MultiHash): Unit
+
+    def put(obj: DataObject): MultiHash = {
```

Comment:
if you change the return type to Multihash you effectively drop References as a type and they are used in quite a lot of places.
Let this return references and encapsulate the multihash.